Expensify/App
# Build a mechanism to migrate Onyx data

<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

### Details
This adds a library to run different Onyx migrations. It also includes a migration file which renames the `activeClients2` key to `activeClients`

One unfortunate side-effect in this particular migration for `activeClients` is that it leaves a key with a `null` value behind:
![image](https://user-images.githubusercontent.com/1228807/110350700-3a7c4280-7ff1-11eb-819f-b59abfa638c5.png)

I think that's OK. I don't really want to add a method to Onyx for deleting a key, and it will get cleared out the next time that they get signed out. Also, it takes up no space so I think it's OK to leave.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify/issues/141785

### Tests
1. Make sure you have `master` checked out
2. Run the application and open the JS console, turn on verbose logging, and filter for logs with "migrate"
3. Check out this branch, and the site should reload
4. Verify the migration runs
5. Refresh the site and verify that the migration is skipped

### Tested On


### Screenshots


#### Web


![image](https://user-images.githubusercontent.com/1228807/110349323-da38d100-7fef-11eb-8bc2-fc3dab9ebe6b.png)

![image](https://user-images.githubusercontent.com/1228807/110349392-e755c000-7fef-11eb-96ac-f65102c657ba.png)

#### Mobile Web

n/a (same as web)

#### Desktop

It's OK that it didn't show the migration running, this was a fresh sign in
![image](https://user-images.githubusercontent.com/1228807/110360485-ffcbd780-7ffb-11eb-8ce0-5f16230599ff.png)

#### iOS

It's OK that it didn't show the migration running, this was a fresh sign in
![image](https://user-images.githubusercontent.com/1228807/110357763-dfe6e480-7ff8-11eb-819c-62d4cd4f7049.png)


#### Android

It's OK that it didn't show the migration running, this was a fresh sign in
![image](https://user-images.githubusercontent.com/1228807/110360009-6c92a200-7ffb-11eb-80c8-7c2ea5fdf481.png)
